vault/identity_store_groups.go

```diff
@@ -346,124 +378,160 @@ func (i *IdentityStore) pathGroupIDDelete() framework.OperationFunc {
 			return logical.ErrorResponse("empty group ID"), nil
 		}
 
-		if groupID == "" {
-			return nil, fmt.Errorf("missing group ID")
-		}
+		return i.handleGroupDeleteCommon(ctx, groupID, true)
+	}
+}
 
-		// Acquire the lock to modify the group storage entry
-		i.groupLock.Lock()
-		defer i.groupLock.Unlock()
+func (i *IdentityStore) pathGroupNameDelete() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+		groupName := d.Get("name").(string)
+		if groupName == "" {
+			return logical.ErrorResponse("empty group name"), nil
+		}
 
-		// Create a MemDB transaction to delete group
-		txn := i.db.Txn(true)
-		defer txn.Abort()
+		return i.handleGroupDeleteCommon(ctx, groupName, false)
+	}
+}
 
-		group, err := i.MemDBGroupByIDInTxn(txn, groupID, false)
-		if err != nil {
-			return nil, err
-		}
+func (i *IdentityStore) handleGroupDeleteCommon(ctx context.Context, key string, byID bool) (*logical.Response, error) {
+	// Acquire the lock to modify the group storage entry
+	i.groupLock.Lock()
+	defer i.groupLock.Unlock()
 
-		// If there is no group for the ID, do nothing
-		if group == nil {
-			return nil, nil
-		}
+	// Create a MemDB transaction to delete group
+	txn := i.db.Txn(true)
+	defer txn.Abort()
 
-		ns, err := namespace.FromContext(ctx)
+	var group *identity.Group
+	var err error
+	switch byID {
+	case true:
+		group, err = i.MemDBGroupByIDInTxn(txn, key, false)
 		if err != nil {
 			return nil, err
 		}
-		if group.NamespaceID != ns.ID {
-			return nil, logical.ErrUnsupportedPath
-		}
-
-		// Delete group alias from memdb
-		if group.Type == groupTypeExternal && group.Alias != nil {
-			err = i.MemDBDeleteAliasByIDInTxn(txn, group.Alias.ID, true)
-			if err != nil {
-				return nil, err
-			}
-		}
-
-		// Delete the group using the same transaction
-		err = i.MemDBDeleteGroupByIDInTxn(txn, group.ID)
+	default:
+		group, err = i.MemDBGroupByNameInTxn(ctx, txn, key, false)
 		if err != nil {
 			return nil, err
 		}
+	}
+	if group == nil {
+		return nil, nil
+	}
+
+	ns, err := namespace.FromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if group.NamespaceID != ns.ID {
+		return nil, logical.ErrUnsupportedPath
```

Comment:
I think we should `return nil, nil` here so it matches the case where there is no group

Reply:
Fixed